### ISSUE TITLE
`solana program deploy` segfaults on macOS

### ISSUE THREAD

**Issue by dmakarov** [solana-labs/solana#33372]
#### Problem

Running `solana program deploy` command on macOS crashes with `Segmentation fault: 11`.  This happens to solana built from source with the current top of the master branch, and only if built in release mode. The same source (of solana cli) built in debug mode works ok.  The deployment is done to a local test validator.

For example:

```
$ solana program deploy --use-quic -k test.json -u localhost program.so
Segmentation fault: 11
```

Or in debugger

```
(lldb) run program deploy --use-quic -k test.json -u localhost program.so
Process 43727 launched: '~/.local/share/solana/install/active_release/bin/solana' (x86_64)
Process 43727 stopped
* thread #1, name = 'main', queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
    frame #0: 0x0000000100d054c8 solana`std::sys::common::thread_local::lazy::LazyKeyInner$LT$T$GT$::initialize::h95cc1ffe6c3fcbe6 + 696
solana`std::sys::common::thread_local::lazy::LazyKeyInner$LT$T$GT$::initialize::h95cc1ffe6c3fcbe6:
->  0x100d054c8 <+696>: movaps %xmm0, 0x150(%rbx)
    0x100d054cf <+703>: movaps -0xb0(%rbp), %xmm0
    0x100d054d6 <+710>: movaps %xmm0, 0x160(%rbx)
    0x100d054dd <+717>: addq   $0x10, %rbx
Target 0: (solana) stopped.
```

#### Proposed Solution

Debug and fix.

**Comment by Lichtso** [comment]
Failed to load / store SIMD registers, which was caused by the stack pointer not being 16 byte aligned.
The fix for that just landed: #33394

**Comment by norwnd** [comment]
Hi @Lichtso, I'm currently observing similar issue on MacOS running latest **master**:
```bash
~/solana (master) » solana program write-buffer /Users/norwnd/solana-random-number-betting-game/solana-program/target/deploy/bpf_program_template.so
[1]    80589 segmentation fault  solana program write-buffer
```
I assume it already includes the fix you mentioned above, hence I'm raising it hoping you can take a look,

I'm quite new to Rust (and solana repo), but I managed to track the culprit down to `fn create_ephemeral_keypair` calling this line (just by using `println!`s and replacing it with hardcoded Mnemonic value):
```rust
let mnemonic = Mnemonic::new(MnemonicType::for_word_count(WORDS)?, Language::English);
```
I also **can't** reproduce this segfault with a simple unit-test:
```rust
    #[test]
    fn test_try_reproduce_segfault() {
        let (_, mnemonic, _) = create_ephemeral_keypair().expect("got error");
        assert_eq!("some phrase", mnemonic.phrase())
    }
```
and I'm not sure what to make of it, thoughts ?
